lib/matplotlib/_pylab_helpers.py

```diff
@@ -71,9 +71,10 @@ def destroy(cls, num):
     @classmethod
     def destroy_fig(cls, fig):
         """Destroy figure *fig*."""
-        canvas = getattr(fig, "canvas", None)
-        manager = getattr(canvas, "manager", None)
-        cls.destroy(manager)
+        num = next((manager.num for manager in cls.figs.values()
+                    if manager.canvas.figure == fig), None)
+        if num is not None:
+            cls.destroy(num)
```

Comment:
Can we add a comment, why we have to destroy by number?

Reply:
It actually works either way (`manager` or `manager.num`); it's just that the code in `destroy` is simpler for `num`.